### PR TITLE
Build: Add back hack for client copying client jar pom

### PR DESCRIPTION
**Pull request by rjernst**
Gradle appears to have a bug in maven publishing which will not match the
artifactId of a generated pom with the artifact id it puts in the file.
This adds back a copy hack from the original pom file name to the client
pom file name (which we had before #20403 inadvertently
removed it).

### CODE REVIEW

**Review by jpountz — APPROVED** [review]

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/plugin/PluginBuildPlugin.groovy [hunk 1]
```diff
@@ -148,12 +148,17 @@ public class PluginBuildPlugin extends BuildPlugin {
     /** Adds a task to move jar and associated files to a "-client" name. */
     protected static void addClientJarTask(Project project) {
         Task clientJar = project.tasks.create('clientJar')
-        clientJar.dependsOn(project.jar, project.javadocJar, project.sourcesJar)
+        clientJar.dependsOn(project.jar, 'generatePomFileForClientJarPublication', project.javadocJar, project.sourcesJar)
```
Comment (danielmitterdorfer) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/plugin/PluginBuildPlugin.groovy]: Shouldn't `generatePomFileForClientJarPublication` be defined as a task somewhere in the project?

Comment (rjernst) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/plugin/PluginBuildPlugin.groovy]: The `maven-publish` plugin generates this task from the publishing configuration.

Comment (danielmitterdorfer) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/plugin/PluginBuildPlugin.groovy]: Ok, didn't know that. Thanks. Then LGTM.

**Review by danielmitterdorfer — COMMENTED** [review]
LGTM.

**Pull request merged**